docs/gallery/domain/plot_icephys.py

```diff
@@ -457,6 +458,38 @@
     category="electrodes",
 )
 
+#####################################################################
+# One predefined subcategory column is the stimulus template column in the stimuli table. This column can be used to
+# store stimulus template waveforms that were generated by the recording software. Similar to the stimulus and response
+# columns, we can specify a relevant time range.
+
+nwbfile.intracellular_recordings.add_column(
+    name="stimulus_template",
+    data=[(0, 5, stimulus),  # (start_index, index_count, stimulus_template) can be specified as a tuple
+          (1, 3, stimulus),
+          (-1, -1, stimulus)],
```

Comment:
I think it may be useful to use `TimeSeriesReference` here, instead of raw tuple to introduce the class and be consistent with how the data represented on read. 

https://github.com/NeurodataWithoutBorders/pynwb/blob/883b1b15810b4a040698c9e40d717f143b3a3248/src/pynwb/base.py#L378

```suggestion
    data=[TimeSeriesReference(0, 5, stimulus),  # (start_index, index_count, stimulus_template) can be specified as a tuple
          TimeSeriesReference(1, 3, stimulus),
          TimeSeriesReference(-1, -1, stimulus)],
```

Reply:
Just a thought, for the empty case ``(-1, -1, stimulus)`` we could maybe add a factory method to `TimeSeriesReference` to create an empty reference, e.g., `TimeSeriesReference.empty()`?

Reply:
The empty reference method sounds helpful. Would we still want a time series (e.g. stimulus) to be required as an input for that method? Or would it create an empty time series with arbitrary units.

Reply:
Good question. I think having an empty `TimeSeries` to point to makes sense. I'm not sure what the best way is to autogenerate that, since we would need to also add it to the file. For now, maybe we can still require that a user to provide the `TimeSeries` and maybe have a best practice on how do to this. Let's discuss when we meet next time how we may be able to do this without the user providing a `TimeSeries`. The `TimeSeries` is necessary because this is stored as compound data type, so all elements must be set, and we don't have a NaN value for ints and object references, so `(-1, -1, <some ref>)` is just a way of saying NaN or invalid. Technically, it really doesn't matter where the reference points to, so maybe there is a way that we can avoid needed a `TimeSeries` at all.